semantic-python/src/Language/Python/Core.hs

```diff
@@ -196,7 +196,22 @@ instance Compile Py.Block where
 
 instance Compile Py.BooleanOperator
 instance Compile Py.BreakStatement
-instance Compile Py.Call
+
+instance Compile Py.Call where
+  compile it@Py.Call
+    { function
+    , arguments = L1 Py.ArgumentList { extraChildren = args }
+    } cc next = do
+    func <- compile function pure next
+    let compileArg = \case
+          Prj expr -> compile (expr :: Py.Expression Span) pure next
+          other   -> fail ("Can't compile non-expression function argument: " <> show other)
```

Comment:
Both the function and its arguments are evaluated in a pure context. The continuation is invoked once the latter have been applied to the former.

Reply:
This is superb.